Unrelated / unchanged files are failing typescript tests on #9582 but passing on develop

<!--
  - Thanks for taking the time to report a bug in the Oppia project.
  - Before filing a new issue, please do a quick search to check that it hasn't
  - already been filed on the [issue tracker](https://github.com/oppia/oppia/issues)._
  -->

**Describe the bug**
#9582 is failing on ```python -m scripts.typescript_checks```, on both local and circleci.

The error messages are as follows ([circleci](https://app.circleci.com/pipelines/github/oppia/oppia/11430/workflows/027c6acd-10f8-4638-8b37-5330af13d26a/jobs/65790/steps)):
```
core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts(565,46): error TS2554: Expected 0 arguments, but got 1.

core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts(587,46): error TS2554: Expected 0 arguments, but got 1.

core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts(609,46): error TS2554: Expected 0 arguments, but got 1.

core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts(631,46): error TS2554: Expected 0 arguments, but got 1.

core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts(653,46): error TS2554: Expected 0 arguments, but got 1.

core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts(675,46): error TS2554: Expected 0 arguments, but got 1.

core/templates/pages/exploration-player-page/exploration-player-page.component.spec.ts(75,7): error TS2554: Expected 0 arguments, but got 1.

core/templates/pages/exploration-player-page/exploration-player-page.component.spec.ts(79,7): error TS2554: Expected 0 arguments, but got 1.

core/templates/pages/exploration-player-page/exploration-player-page.component.spec.ts(84,7): error TS2554: Expected 0 arguments, but got 1.

core/templates/pages/exploration-player-page/exploration-player-page.component.spec.ts(88,7): error TS2554: Expected 0 arguments, but got 1.
```

These 2 files (`exploration-player-page.component.spec.ts`, `exploration-editor-page.component.spec.ts`) are completely unchanged in the PR, however. 

All of the errors are related to the below usage:
```
angularElementSpy.withArgs('meta[itemprop="name"]')
```
with error:
```
Expected 0 arguments, but got 1.
```


Note: A similar error occurred on a different file in the PR (extensions/interactions/GraphInput/directives/graph-viz.directive.ts), and had to be manually resolved. Again, this file or any files referencing it was not changed.


**To Reproduce**
Steps to reproduce the behavior:
 1. Run ```python -m scripts.typescript_checks``` on #9582

--- a/core/templates/components/button-directives/exploration-embed-button-modal.controller.spec.ts
+++ b/core/templates/components/button-directives/exploration-embed-button-modal.controller.spec.ts
@@ -52,7 +52,9 @@ describe('Exploration Embed Button Modal Controller', function() {
   it('should add range from a click event', function() {
     var removeAllRanges = jasmine.createSpy('removeAllRanges');
     var addRange = jasmine.createSpy('addRange');
-    spyOn(window, 'getSelection').and.returnValue(<any>{
+    // @ts-ignore window getSelection method return more properties than
+    // removeAllRanges and addRange.
+    spyOn(window, 'getSelection').and.returnValue({
       removeAllRanges: removeAllRanges,
       addRange: addRange
     });

--- a/core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts
+++ b/core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts
@@ -562,8 +562,12 @@ describe('Exploration editor page component', function() {
           // should be declared.
           spyOn(window, '$').and.returnValue(element);
           var animateSpy = spyOn(element, 'animate').and.callThrough();
+          // @ts-ignore Angular element method doesn't expect to receive
+          // 1 argument in the lints.
           spyOn(angular, 'element').withArgs('#tutorialStateContent').and
-            .returnValue(<any>{
+            .returnValue({
+              // @ts-ignore Angular element should have more properties than
+              // just offset in the lint settings.
               offset: () => ({
                 top: 5
               })
@@ -584,8 +588,12 @@ describe('Exploration editor page component', function() {
           // should be declared.
           spyOn(window, '$').and.returnValue(element);
           var animateSpy = spyOn(element, 'animate').and.callThrough();
+          // @ts-ignore Angular element method doesn't expect to receive
+          // 1 argument in the lints.
           spyOn(angular, 'element').withArgs('#tutorialStateInteraction').and
-            .returnValue(<any>{
+            .returnValue({
+              // @ts-ignore Angular element should have more properties than
+              // just offset in the lint settings.
               offset: () => ({
                 top: 20
               })
@@ -606,8 +614,12 @@ describe('Exploration editor page component', function() {
           // should be declared.
           spyOn(window, '$').and.returnValue(element);
           var animateSpy = spyOn(element, 'animate').and.callThrough();
+          // @ts-ignore Angular element method doesn't expect to receive
+          // 1 argument in the lints.
           spyOn(angular, 'element').withArgs('#tutorialPreviewTab').and
-            .returnValue(<any>{
+            .returnValue({
+              // @ts-ignore Angular element should have more properties than
+              // just offset in the lint settings.
               offset: () => ({
                 top: 5
               })
@@ -628,8 +640,12 @@ describe('Exploration editor page component', function() {
           // should be declared.
           spyOn(window, '$').and.returnValue(element);
           var animateSpy = spyOn(element, 'animate').and.callThrough();
+          // @ts-ignore Angular element method doesn't expect to receive
+          // 1 argument in the lints.
           spyOn(angular, 'element').withArgs('#tutorialStateInteraction').and
-            .returnValue(<any>{
+            .returnValue({
+              // @ts-ignore Angular element should have more properties than
+              // just offset in the lint settings.
               offset: () => ({
                 top: 20
               })
@@ -650,8 +666,12 @@ describe('Exploration editor page component', function() {
           // should be declared.
           spyOn(window, '$').and.returnValue(element);
           var animateSpy = spyOn(element, 'animate').and.callThrough();
+          // @ts-ignore Angular element method doesn't expect to receive
+          // 1 argument in the lints.
           spyOn(angular, 'element').withArgs('#tutorialPreviewTab').and
-            .returnValue(<any>{
+            .returnValue({
+              // @ts-ignore Angular element should have more properties than
+              // just offset in the lint settings.
               offset: () => ({
                 top: 5
               })
@@ -672,8 +692,12 @@ describe('Exploration editor page component', function() {
           // should be declared.
           spyOn(window, '$').and.returnValue(element);
           var animateSpy = spyOn(element, 'animate').and.callThrough();
+          // @ts-ignore Angular element method doesn't expect to receive
+          // 1 argument in the lints.
           spyOn(angular, 'element').withArgs('#tutorialStateInteraction').and
-            .returnValue(<any>{
+            .returnValue({
+              // @ts-ignore Angular element should have more properties than
+              // just offset in the lint settings.
               offset: () => ({
                 top: 20
               })

--- a/core/templates/pages/exploration-player-page/exploration-player-page.component.spec.ts
+++ b/core/templates/pages/exploration-player-page/exploration-player-page.component.spec.ts
@@ -72,21 +72,29 @@ describe('Exploration player page', function() {
 
     var elementNameItemProp = $('<div>');
     angularElementSpy.withArgs(
-      'meta[itemprop="name"]').and.returnValue(<any>elementNameItemProp);
+      // @ts-ignore Angular element method doesn't expect to receive 1 argument
+      // in the lints.
+      'meta[itemprop="name"]').and.returnValue(elementNameItemProp);
 
     var elementDescriptionItemProp = $('<div>');
     angularElementSpy.withArgs(
+      // @ts-ignore Angular element method doesn't expect to receive 1 argument
+      // in the lints.
       'meta[itemprop="description"]').and.returnValue(
-        <any>elementDescriptionItemProp);
+      elementDescriptionItemProp);
 
     var elementTitleProperty = $('<div>');
     angularElementSpy.withArgs(
-      'meta[property="og:title"]').and.returnValue(<any>elementTitleProperty);
+      // @ts-ignore Angular element method doesn't expect to receive 1 argument
+      // in the lints.
+      'meta[property="og:title"]').and.returnValue(elementTitleProperty);
 
     var elementDescriptionProperty = $('<div>');
     angularElementSpy.withArgs(
+      // @ts-ignore Angular element method doesn't expect to receive 1 argument
+      // in the lints.
       'meta[property="og:description"]').and.returnValue(
-        <any>elementDescriptionProperty);
+      elementDescriptionProperty);
 
     ctrl.$onInit();
     $scope.$apply();

--- a/core/templates/pages/preferences-page/modal-templates/edit-profile-picture-modal.controller.spec.ts
+++ b/core/templates/pages/preferences-page/modal-templates/edit-profile-picture-modal.controller.spec.ts
@@ -88,7 +88,9 @@ describe('EditProfilePictureModalController', function() {
         'data:application/octet-stream;base64,' + dataBase64Mock);
 
       var mockUrl = 'mock-url';
-      spyOn(Cropper.prototype, 'getCroppedCanvas').and.returnValue(<any>{
+      // @ts-ignore Cropper getCroppedCanvas method should return more
+      // properties than toDataURL according with lint settings.
+      spyOn(Cropper.prototype, 'getCroppedCanvas').and.returnValue({
         toDataURL: () => mockUrl
       });
       $scope.confirm();

--- a/extensions/interactions/LogicProof/static/js/tools/logic-demo-test.controller.spec.ts
+++ b/extensions/interactions/LogicProof/static/js/tools/logic-demo-test.controller.spec.ts
@@ -476,7 +476,9 @@ describe('Logic demo test', function() {
 
   it('should submit mistakes', function() {
     const MISTAKE_STRINGS_LENGTH = 40;
-    spyOn(logicProofTeacher2, 'buildMistakeSection').and.returnValue(<any>[{
+    // @ts-ignore logicProofTeacher2 buildMistakeSection method should return
+    // ore properties than entries according with lint settings.
+    spyOn(logicProofTeacher2, 'buildMistakeSection').and.returnValue([{
       entries: []
     }]);
     var sectionNumber = 0;
@@ -500,8 +502,9 @@ describe('Logic demo test', function() {
   });
 
   it('should submit control functions', function() {
-    spyOn(logicProofTeacher2, 'buildControlFunctionTable').and.returnValue(
-      <any>{});
+    // @ts-ignore logicProofTeacher2 buildControlFunctionTable method should
+    // return more properties than {} according with lint settings.
+    spyOn(logicProofTeacher2, 'buildControlFunctionTable').and.returnValue({});
     $scope.submitControlFunctions();
 
     expect($scope.controlFunctionSuccess).toBe(true);
@@ -536,11 +539,14 @@ describe('Logic demo test', function() {
   });
 
   it('should request javascript', function() {
-    spyOn(logicProofTeacher2, 'buildMistakeSection').and.returnValue(<any>[{
+    // @ts-ignore logicProofTeacher2 buildMistakeSection method should return
+    // ore properties than entries according with lint settings.
+    spyOn(logicProofTeacher2, 'buildMistakeSection').and.returnValue([{
       entries: []
     }]);
-    spyOn(logicProofTeacher2, 'buildControlFunctionTable').and.returnValue(
-      <any>{});
+    // @ts-ignore logicProofTeacher2 buildControlFunctionTable method should
+    // return more properties than {} according with lint settings.
+    spyOn(logicProofTeacher2, 'buildControlFunctionTable').and.returnValue({});
 
     $scope.submitMistakes(0);
     $scope.submitMistakes(1);

